falcon/testing/client.py

```diff
@@ -749,3 +1140,49 @@ def simulate_request(self, *args, **kwargs):
             kwargs['headers'] = merged_headers
 
         return simulate_request(self.app, *args, **kwargs)
+
+
+# -----------------------------------------------------------------------------
+# Private
+# -----------------------------------------------------------------------------
+
+
+def _is_asgi_app(app):
+    app_args = inspect.getfullargspec(app).args
+    num_app_args = len(app_args)
+
+    # NOTE(kgriffs): Technically someone could name the "self"
+    #   arg something else, but we will make the simplifying
+    #   assumption that this is rare enough to not worry about.
+    if app_args[0] == 'self':
```

Comment:
As a fan of `@classmethod`s, would that be supported OoB, or do we need to add support for `app_args[0] == 'cls'`?
I would vote in favour in that case :slightly_smiling_face: